pkg/security/security_profile/activity_tree/activity_tree.go

```diff
@@ -277,12 +277,12 @@ func (at *ActivityTree) Insert(event *model.Event, generationType NodeGeneration
 
 // Contains looks up the event in the activity tree
 func (at *ActivityTree) Contains(event *model.Event, generationType NodeGenerationType, resolvers *resolvers.Resolvers) (bool, error) {
-	newEntry, err := at.insertEvent(event, true, generationType, resolvers)
+	newEntry, err := at.insertEvent(event, true, false, generationType, resolvers)
```

Comment:
Not a blocker, but when you start having multiple boolean options for a function it starts loosing its simplicity. Maybe you can provide a struct or something similar

Reply:
Yeah I agree with you. I was feeling the same adding it, but as this func is only called twice on the Insert/Contains helpers just above I assumed it was kinda ok. I've just added some comments to make those calls less blurry.